components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTx.java

```diff
@@ -88,10 +88,22 @@
   Iterable<Component> browseComponents(Bucket bucket);
 
   /**
+   * Gets a bucket by id, or {@code null} if not found.
+   */
+  @Nullable
+  Bucket findBucket(EntityId id);
+
+  /**
+   * Gets an asset by id, or {@code null} if not found.
+   */
+  @Nullable
+  Asset findAsset(EntityId id);
+
+  /**
    * Gets an asset by id, owned by the specified bucket, or {@code null} if not found.
    */
   @Nullable
-  Asset findAsset(EntityId id, Bucket bucket);
+  Asset findBucketAsset(EntityId id, Bucket bucket);
```

Comment:
I'd prefer to keep `findAsset` as the method name, since that's what it returns - the javadoc and extra argument already tells me that this is finding the asset belonging to the given bucket.